src/server/terrain/structure_building_blocks.zig

```diff
@@ -0,0 +1,149 @@
+const std = @import("std");
+
+const main = @import("root");
+const ZonElement = main.ZonElement;
+const Blueprint = main.blueprint.Blueprint;
+const List = main.List;
+const ListUnmanaged = main.ListUnmanaged;
+const AliasTable = main.utils.AliasTable;
+const Neighbor = main.chunk.Neighbor;
+const Block = main.blocks.Block;
+const Degrees = main.rotation.Degrees;
+const NeverFailingAllocator = main.heap.NeverFailingAllocator;
+
+var arena = main.heap.NeverFailingArenaAllocator.init(main.globalAllocator);
+const arenaAllocator = arena.allocator();
+
+var blueprintCache: std.StringHashMapUnmanaged(*[4]BlueprintEntry) = .{};
+var childrenToResolve: List(struct {parentId: []const u8, colorName: []const u8, colorIndex: usize, childIndex: usize, structureId: []const u8}) = undefined;
+
+const originBlockStringId = "cubyz:sbb/origin";
+var originBlockNumericId: u16 = 0;
+
+// Maps global child block numeric ID to index used to locally represent that child block.
+var childBlockNumericIdMap: std.AutoHashMapUnmanaged(u16, u16) = .{};
+var childBlockStringId: ListUnmanaged([]const u8) = .{};
```

Comment:
Just for my own understanding: What is this needed for again? Was it for printing nicer log messages?

Reply:
In configuration files you specify what should spawn at given child block by specifying color name of that child block and since we load list of child blocks dynamically, string IDs are used to determine color names. [Example](https://github.com/Argmaster/Cubyz/blob/feature/SBBGen/assets/cubyz/sbb/tree/oak/stem0.zig.zon)

Reply:
Shouldn't it be a hashmap then, so you can lookup the value without needing to iterate the list?

Reply:
No, indexes of IDs match `StructureBlock.index` and can be obtained through mapping block typ to index with `childBlockNumericIdMap`

Reply:
I suppose it will make more sense after #1240

Reply:
Ah I see